dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlannerExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Planning.Sequential;
+
+/// <summary>
+/// Extension methods for <see cref="SequentialPlanner"/> class.
+/// </summary>
+public static class SequentialPlannerExtensions
+{
+    /// <summary>
+    /// Returns decorated instance of <see cref="ISequentialPlanner"/> with enabled instrumentation.
+    /// </summary>
+    /// <param name="planner">Instance of <see cref="ISequentialPlanner"/> to decorate.</param>
+    /// <param name="logger">Optional logger.</param>
+    public static ISequentialPlanner WithInstrumentation(this ISequentialPlanner planner, ILogger? logger = null)
```

Comment:
WithLogger?

Reply:
Even though this method accepts optional logger, it returns instance of `InstrumentedSequentialPlanner`, which supports not only logs, but also metrics and traces. You don't need to pass metering or tracing-related objects here, because you can subscribe to the telemetry events directly using event names or by using methods `WithMetering` and `WithTracing` in `KernelBuilder`, which will subscribe to all Semantic Kernel telemetry events.